sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Primitives/MessagingProvider.cs

```diff
@@ -16,210 +18,119 @@ namespace Microsoft.Azure.WebJobs.ServiceBus
     public class MessagingProvider
```

Comment:
Can we merge provider and client factory?

Reply:
I intentionally split these to keep the MessagingProvider ctor free of dependencies on Config/ComponentFactory.